### DISCUSSION TITLE
How to get count of hits by response.status_code

### DISCUSSION THREAD

**Discussion by dbire** [schemathesis/schemathesis#1942] [Q&A]
Hello, 

I am using the CLI version and I am wondering how it is possible to print at the end of the st run the  count of hits by response.status_code to check if I was able to cover the status code expected ?  

For example, it could happen that st runs and find no failure just because it gets only 400 or 404, and found no errors regarding the conformity... and 20x checks would be under the radar.

**Comment by Stranger6667** [comment]
Hi, it is not implemented, but requested, see #1414. Schemathesis collects this data, but we need to display it nicely and I don’t know what would be the best way to do so

**Reply by dbire** [reply]
For example,  @Stranger6667  what do you think of the following ?

ENDPOINT       METHOD    STATUS   #HITS
------------      -----------    --------- -------- 
/users               post              200        50
/users               post              400        10
/users/login      post              200        20
....

**Reply by Stranger6667** [reply]
The main issue is that it may work for relatively small cases, but for e.g 70 endpoints with 5-6 statuses each it will not be readable and will draw attention from other parts of the output, ie 300 lines of text appearing at the end of CLI output is quite distracting. Even smaller would be problematic, e.g often we see 200 & 400 only, so for 10 endpoints it will be 10-20 lines, but I find the most APIs in the wild are much larger than that

**Reply by Stranger6667** [reply]
Drawing a table might be better though, so the data is more compact

**Reply by dbire** [reply]
I have written a small python script that uses  a generated "cassette.yml" as input and give me some statistics of what have been covered  If anybody is interested...

**Reply by ludeknovy** [reply]
Hi @Stranger6667 !
Is this data somehow accessible from the python library API? I would like to amend a custom check, that endpoint not returned only 401/403 - which might happen in case of wrong credentials and would stay under the radar (the test would pass).

**Reply by Stranger6667** [reply]
Hi @ludeknovy 

Not right now! The `pytest` integration only has access to the current test scope - results of any API calls made in a test body aren't stored anywhere. I think that right now the only way would be to manually store them in some container and run those checks outside of that test (i.e. in some "dependent" tests as some pytest plugins provide). I have a plan to implement custom checks with a wider scope, but I haven't planned the `pytest` integration aspect yet.

**Reply by ludeknovy** [reply]
Thanks for the hints. I will try to implement it using the pytest plugin.
